File: src/main/java/practice/FindIndexKNode.java
package practice;

import java.util.ArrayList;

public class FindIndexKNode {
    public static void main(String[] args) {
        int[] arr = {1,2,3,4,5,6,7,8};
        System.out.println(makeLinkedByArray(arr).element);
        System.out.println(makeLinkedByArray(arr).nextNode.element);
    }
    // 返回以数组创建一条链表的尾节点
    public static LinkedNode makeLinkedByArray(int[] arr) {
        LinkedNode first = new LinkedNode(-1);
        LinkedNode last = first;

        for (int i: arr) {
            LinkedNode now = new LinkedNode(i);
            last.nextNode = now;
            last = now;

        }

        return  first;
    }


}

File: src/main/scala/linked/practice/FindIndexKNode.scala
package linked.practice

/**
 * 输入一个链表，输出该链表中倒数第k个结点。
 * 链表数据结构面试题-多家公司
 */
object FindIndexKNode {

  // 返回以数组创建一条链表的头节点
  def makeLinkedByArray(list: List[Int]): LinkedNode = {
    val node = new LinkedNode(-1)
    list.foreach(x => {
      val next = new LinkedNode(x)
      node.next = next
    })
    node
  }

}

case class LinkedNode(element: Int, nextNode: LinkedNode){
  var elem: Int = element
  var next: LinkedNode = nextNode

  def this(element: Int){
    this(element, null)
  }

}
